examples/sudt-transfer.rs

```diff
@@ -443,101 +443,106 @@ async fn receive(
-                .build_exact_capacity(Capacity::bytes(16).unwrap())
-                .unwrap(),
+        .await?;
+        let channel_contract_cell = get_latest_cell_by_type_script(
+            &client,
+            config.sdk_config.channel_contract_type_script().into(),
         )
-        .output_data(sudt_amount_data(pd.amount.into()).pack())
-        // capacity input and witness.
-        .input(simple_input(user_input.out_point.into()))
-        .witness(placeholder_witness.as_bytes().pack());
+        .await?;
+        let packet_contract_cell = get_latest_cell_by_type_script(
+            &client,
+            config.sdk_config.packet_contract_type_script().into(),
+        )
+        .await?;
+        let channel = IbcChannelCell::get_latest(&client, &config.sdk_config).await?;
+
+        let (tx, envelope) = assemble_write_ack_partial_transaction(
+            simple_dep(axon_metadata_cell.out_point.into()),
+            simple_dep(channel_contract_cell.out_point.into()),
+            simple_dep(packet_contract_cell.out_point.into()),
+            &config.sdk_config,
+            channel,
+            p,
+            vec![1],
+        )?;
+
+        let user_input = get_capacity_input(&client, &user_lock_script).await?;
+
+        // sighash placeholder witness.
+        let placeholder_witness = packed::WitnessArgs::new_builder()
+            .lock(Some(Bytes::from_static(&[0u8; 65])).pack())
+            .build();
+        let tx = tx
+            // st-cell input/output.
+            .input(simple_input(st_cell.out_point.into()))
+            .output(packed::CellOutput::from(st_cell.output))
+            .output_data(
+                sudt_amount_data(
+                    st_cell_amount
+                        .checked_sub(pd.amount.into())
+                        .context("st-cell amount not enough")?,
+                )
+                .pack(),
+            )
+            .witness([].pack())
+            .cell_dep(sudt_transfer_dep.clone())
+            .cell_dep(sudt_dep)
+            // sudt output.
+            .output(
+                packed::CellOutput::new_builder()
+                    .lock(user_lock_script.clone())
+                    .type_(Some(sudt_type_script).pack())
+                    .build_exact_capacity(Capacity::bytes(16).unwrap())
+                    .unwrap(),
+            )
+            .output_data(sudt_amount_data(pd.amount.into()).pack())
+            // capacity input and witness.
+            .input(simple_input(user_input.out_point.into()))
+            .witness(placeholder_witness.as_bytes().pack());
 
-    let tx = add_ibc_envelope(tx, &envelope).build();
+        let tx = add_ibc_envelope(tx, &envelope).build();
 
-    let tx = complete_tx(&config.ckb_rpc_url, &tx, user_lock_script, sk)?;
+        let tx = complete_tx(&config.ckb_rpc_url, &tx, user_lock_script.clone(), sk)?;
 
-    send_transaction(&config.ckb_rpc_url, tx)?;
+        send_transaction(&config.ckb_rpc_url, tx)?;
```

Comment:
can send these packets searched in a batch, like 5 of them, in one transaction?

Reply:
I don't understand your question/comment.

Reply:
> I don't understand your question/comment.

I mean, one transaction contains multi RecvPacket cells

Reply:
You mean the current code will do this and it shouldn't? I don't think it will.

Or do you mean it should send packets in batch? I don't think it's possible.